File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.java
class Solution {
    public int findMin(int[] nums) {
        int r = nums.length-1;
        int l =0;
        while(l<r){
            int mid = (l+r)/2;
            if(nums[mid]<nums[r]){
                r=mid;
            }else{
                l=mid+1;
            }
        }

        return nums[l];
    }
}

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-2.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        length= float("inf")
        l , n = 0 , len(nums)
        curSum=nums[l]
        if curSum>=target:
            return 1
        for r in range(1,n):
            curSum+=nums[r]

            while curSum>=target and l<=r:
                length=min(length,r-l+1)
                curSum-=nums[l]
                l+=1
        return length if length!=float("inf") else 0